Modules/IO/NIFTI/test/itkNiftiImageIOTest14.cxx

```diff
@@ -0,0 +1,104 @@
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkNiftiImageIOTest.h"
+#include "itkTestingMacros.h"
+
+// Test writing NIFTI xyzt_units field - should be 10 (mm, sec)
+int
+itkNiftiImageIOTest14(int argc, char * argv[])
+{
+
+  if (argc > 1)
+  {
+    char * testdir = *++argv;
+    itksys::SystemTools::ChangeDirectory(testdir);
+  }
+  else
+  {
+    return EXIT_FAILURE;
+  }
+  int success(0);
+
+  constexpr unsigned int Dimension = 3;
+
+  using PixelType = float;
+  using ImageType = itk::Image<PixelType, Dimension>;
+
+  ImageType::Pointer image;
+
+  // fill out an image and write it as nifti.
+  const char * filename = "xyzt_unit_check.nii";
+  using OutputImageType = itk::Image<PixelType, Dimension>;
+  typename OutputImageType::RegionType region;
+  typename OutputImageType::IndexType  start;
+  start[0] = 0;
+  start[1] = 0;
+  start[2] = 0;
+  typename OutputImageType::SizeType size;
+  size[0] = 8;
+  size[1] = 8;
+  size[2] = 4;
+  region.SetSize(size);
+  region.SetIndex(start);
+  auto outputimage = OutputImageType::New();
+  outputimage->SetRegions(region);
+  outputimage->Allocate();
+
+  using WriterType = itk::ImageFileWriter<OutputImageType>;
+  auto                       writer = WriterType::New();
+  itk::NiftiImageIO::Pointer niftiImageIO(itk::NiftiImageIO::New());
+  writer->SetImageIO(niftiImageIO);
+  writer->SetFileName(filename);
+  writer->SetInput(outputimage);
+  try
+  {
+    writer->Update();
+  }
+  catch (const itk::ExceptionObject & err)
+  {
+    std::cerr << "Exception Object caught: " << std::endl << err << std::endl;
+    throw;
+  }
+
+  try
+  {
+    // read the image back in
+    ImageType::Pointer image_from_disk = itk::IOTestHelper::ReadImage<ImageType>(std::string(filename));
+    // check the metadata for xyzt_units
+    itk::MetaDataDictionary & dictionary = image_from_disk->GetMetaDataDictionary();
+    std::string               xyzt_units;
+    if (!itk::ExposeMetaData<std::string>(dictionary, "xyzt_units", xyzt_units))
+    {
+      std::cerr << "xyzt_units not found in metadata" << std::endl;
+      return EXIT_FAILURE;
+    }
+    if (xyzt_units != "10")
+    {
+      std::cerr << "xyzt_units not set correctly in metadata" << std::endl;
+      return EXIT_FAILURE;
+    }
+  }
+  catch (...)
+  {
+    itk::IOTestHelper::Remove(filename);
+    return EXIT_FAILURE;
```

Comment:
Thanks for your PR, @cookpa Looking at these lines of code, I was wondering, should this file ("xyzt_unit_check.nii") _only_ be removed in case of a failure (`EXIT_FAILURE`)? Or also in the EXIT_SUCCESS case?

Reply:
You're right, I'll fix this so it always gets removed